### PR TITLE
Add CBMC proof of Free_FreeRTOS function

### PR DESCRIPTION
**Pull request by Bujain**
Add CBMC proof of Free_FreeRTOS function

Description
-----------
This PR adds files required for the CBMC proof of free_FreeRTOS function. 

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is formatted using Uncrustify.
- [x] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yanjos-dev** [comment]
What version of the kernel is the submodule set to? It should be on the latest release and not on master.

**Review by divekarshubham — APPROVED** [review]

**Review by divekarshubham — APPROVED** [review]

File: test/cbmc/proofs/Free_FreeRTOS/Free_FreeRTOS_harness.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * @file Free_FreeRTOS_harness.c
+ * @brief Implements the proof harness for Free_FreeRTOS function.
+ */
+/*  FreeRTOS includes for OTA library. */
+#include "ota_os_freertos.h"
+
+/* Stub to free dynamically allocated memory in FreeRTOS. */
+int vPortFree( void * ptr )
+{
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/Free_FreeRTOS/Free_FreeRTOS_harness.c:32]: Do we want to asset that ptr should not be NULL?

Comment (AniruddhaKanhere) [reply test/cbmc/proofs/Free_FreeRTOS/Free_FreeRTOS_harness.c:32]: I am not sure whether FreeRTOS vPortFree allows a NULL to be freed. Might be worth looking into it.

File: test/cbmc/proofs/Free_FreeRTOS/Free_FreeRTOS_harness.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * @file Free_FreeRTOS_harness.c
+ * @brief Implements the proof harness for Free_FreeRTOS function.
+ */
+/*  FreeRTOS includes for OTA library. */
+#include "ota_os_freertos.h"
+
+/* Stub to free dynamically allocated memory in FreeRTOS. */
+int vPortFree( void * ptr )
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/Free_FreeRTOS/Free_FreeRTOS_harness.c:31]: Is the return datatype of vPortFree an `int`? I do remember that it was `void`?

**Review by AniruddhaKanhere — COMMENTED** [review]
Hello @Bujain, would you mind looking at the comments/questions that I have?

**Comment by AniruddhaKanhere** [comment]
Also, maybe try allocating an actual pointer (using malloc) and then free it as well.

**Comment by Bujain** [comment]
> 
> 
> Also, maybe try allocating an actual pointer (using malloc) and then free it as well.

Since we are not checking the memory safety of the external library functions. It does not make sense to do this.

**Review by AniruddhaKanhere — APPROVED** [review]

**Pull request merged**